include/fs/datablock_manager.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef DATABLOCK_MANAGER_HPP
+#define DATABLOCK_MANAGER_HPP
+
+#include "fs_constants.hpp"
+
+class Fs;
+
+class DataBlock_Manager {
+public:
+  DataBlock_Manager(Fs *fs, u_int64_t block_segment_start,
+                      u_int64_t block_segment_end);
+
+  virtual int new_datablock(u_int64_t *block_num) = 0;
+  virtual int free_datablock(u_int64_t block_num) = 0;
+
+  virtual int format() = 0;
+
+protected:
+  Fs *fs;
+  u_int64_t block_segment_start, block_segment_end;

```

Comment:
Does block_segment_end mean the start address of the end of a group of 2048 datablocks managed by a bitmap block?

Reply:
No, this is for the datalock_manager to use for controlling the entire datablock region. It defines the start and end of the datablock region on disk, the start being inclusive and the end being non-inclusive.